packages/alpinejs/src/magics/$watch.js

```diff
@@ -17,10 +17,10 @@ magic('watch', (el, { evaluateLater, effect }) => (key, callback) => {
             queueMicrotask(() => {
                 callback(value, oldValue)
 
-                oldValue = value
+                oldValue = JSON.parse(JSON.stringify(value))
```

Comment:
JSON.stringify is expensive, the code already does it at line 12 so you can just assign it to a temp variable and reuse that in here?